eox_nelp/api_clients/tests/tests_certificates.py

```diff
@@ -0,0 +1,65 @@
+"""This file contains all the test for certificates api client file.
+
+Classes:
+    TestExternalCertificatesApiClient: Test for eox-nelp/api_clients/certificates.py.
+"""
+import unittest
+
+from django.utils import timezone
+from mock import patch
+
+from eox_nelp.api_clients.certificates import ExternalCertificatesApiClient
+from eox_nelp.api_clients.tests import BasicApiClientMixin
+
+
+class TestExternalCertificatesApiClient(BasicApiClientMixin, unittest.TestCase):
+    """Tests ExternalCertificatesApiClient"""
+
+    def setUp(self):
+        """Setup common conditions for every test case"""
+        self.api_class = ExternalCertificatesApiClient
+
+    @patch.object(ExternalCertificatesApiClient, "make_post")
+    @patch.object(ExternalCertificatesApiClient, "_authenticate")
+    def test_create_certificate(self, auth_mock, post_mock):
+        """Test successful post request.
+
+        Expected behavior:
+            - Response is the expected value
+        """
+        auth_mock.return_value = {}
+        expected_value = {
+            "status": {"success": True, "message": "successful", "code": 1}
+        }
+        post_mock.return_value = expected_value
+        user = {
+            "national_id": "10224587",
+            "english_name": " Testing",
+            "arabic_name": "اختبارات",
+        }
+        data = {
+            "id": "124ABC",
+            "created_at": timezone.now(),
+            "expiration_date": timezone.now() + timezone.timedelta(days=365),
+            "grade": 10,
+            "is_passing": True,
+            "user": user,
+        }
+        api_client = self.api_class()
+
+        response = api_client.create_external_certificate(data)
+
+        self.assertDictEqual(response, expected_value)
+
+    @patch.object(ExternalCertificatesApiClient, "_authenticate")
+    def test_failed_create_certificate(self, auth_mock):
+        """Test when the mandatory fields has not been sent.
+
+        Expected behavior:
+            - Raise KeyError exception.
+        """
+        auth_mock.return_value = {}
+        data = {}
+        api_client = ExternalCertificatesApiClient()
+
+        self.assertRaises(KeyError, api_client.create_external_certificate, data)
```

Comment:
If the auth is a mock, the API client first raised is a key error of dict?

Reply:
probably I dont understand your question the auth_mock helps to skip errors when the client is initialized  line 63 and this test is for the execution of `create_external_certificate`

Reply:
Sure, is strange that the auth is also mocked. Then I viewed the successful test and the auth is mocked too.

Reply:
The authentication happens when the class is initialized  if the initialization fails the class instance doesn't exists therefore the method can not be called  and these tests can not be performed, and keep in mind these tests are not related with the authentication